### DISCUSSION TITLE
Composite Texture - Combining grayscale textures into one RGB(A) image

### DISCUSSION THREAD

**Discussion by Braveo** [godotengine/godot-proposals#5867] [General Discussions]
So I saw Decals that use ORM as one texture to compress file size, but what if there was a way to get one resource to be able to put together an image based on a few other images? That way, we can stack different maps together in order without having to go to some external image editing software. Maybe in compile time, it could be packed into one singular image? I'm not sure, just a thought.

**Comment by QbieShay** [comment]
Yes please

I think it could be a special type of ImageTexture that accepts 1-4 images and per each image it has a property that says from which channel to pull from. However, this could be done with an addon first.

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot-proposals/issues/2316.

The idea of doing this at run-time with a CompositeTexture resource type is interesting (rather than an editor tool as I proposed). However, I wonder if this will cause issues with VRAM-compressed textures or slow down loading times when using large textures this way.
